src/blindpsbt.cpp

```diff
@@ -400,83 +400,40 @@ BlindingStatus BlindPSBT(PartiallySignedTransaction& psbt, std::map<uint32_t, st
                     GenerateAssetEntropy(entropy, input.GetOutPoint(), input.m_issuance_asset_entropy);
                 }
 
-                // Asset isn't blinded yet. Add it to the list of input assets
-                CalculateAsset(issuance_asset, entropy);
-                fixed_input_tags.emplace_back();
-                memcpy(fixed_input_tags.back().data, issuance_asset.begin(), 32);
-                ephemeral_input_tags.emplace_back();
-                if (secp256k1_generator_generate(secp256k1_blind_context, &ephemeral_input_tags.back(), issuance_asset.begin()) != 1) {
-                    return BlindingStatus::INVALID_ASSET;
+                if (input.m_issuance_value != std::nullopt || input.m_issuance_value_commitment.IsCommitment()) {
+                    // Asset isn't blinded yet. Add it to the list of input assets
+                    CalculateAsset(issuance_asset, entropy);
+                    fixed_input_tags.emplace_back();
+                    memcpy(fixed_input_tags.back().data, issuance_asset.begin(), 32);
+                    ephemeral_input_tags.emplace_back();
+                    if (input.m_issuance_value_commitment.IsNull()) {
+                        if (secp256k1_generator_generate(secp256k1_blind_context, &ephemeral_input_tags.back(), issuance_asset.begin()) != 1) {
+                            return BlindingStatus::INVALID_ASSET;
+                        }
+                    }
+                    else {
+                        memcpy(ephemeral_input_tags.back().data, input.m_issuance_value_commitment.vchCommitment.data(), 33);
+                    }
+                    input_asset_blinders.emplace_back();
                 }
-                unsigned int iss_to_blind = 1; // Always do the first issuance blinding iteration for the issuance value
 
-                bool blind_issuance = our_issuances_to_blind.count(i) > 0;
+                bool blind_issuance = input.m_issuance_inflation_keys_commitment.IsCommitment() || input.m_issuance_value_commitment.IsCommitment();
 
-                if (input.m_issuance_blinding_nonce.IsNull() && input.m_issuance_inflation_keys_amount) {
+                if (input.m_issuance_blinding_nonce.IsNull() && (input.m_issuance_inflation_keys_amount != std::nullopt || input.m_issuance_inflation_keys_commitment.IsCommitment())) {
```

Comment:
Where are we dealing with the case `m_issuance_blinding_nonce` is not Null? Seems like this would not work correctly in re-issuance case?

Reply:
In a reissuance case, there are no reissuance tokens being created, only the asset tokens. Those are handed in the section above.